arrow-array/src/array/mod.rs

```diff
@@ -173,52 +173,60 @@ pub trait Array: std::fmt::Debug + Send + Sync {
     /// ```
     fn offset(&self) -> usize;
 
-    /// Returns the null buffer of this array if any
+    /// Returns the null buffer of this array if any.
     ///
-    /// Note: some arrays can encode their nullability in their children, for example,
+    /// The null buffer encodes the "physical" nulls of an array and is pre-computed and very efficient.
+    /// However, some arrays can also encode nullability in their children, for example,
```

Comment:
```suggestion
    /// The null buffer encodes the "physical" nulls of an array.
    /// However, some arrays can also encode nullability in their children, for example,
```

I think the notion of it being pre-computed is a bit confusing, it isn't really computed at all, you wouldn't describe the values as being pre-computed?